### PR TITLE
Fix IRFFT contrib op output dimension calculation

### PR DESCRIPTION
**Pull request by Alexey-Kamenev**
### Description
Fixes the issue with IRFFT output dimension calculation as described in #13236 

### Motivation and Context
Please refer to #13236 for detailed description.

### CODE REVIEW

**Comment by justinchuby** [comment]
I think this PR needs a rebase. Could you do that?

**Comment by NickGeneva** [comment]
> I think this PR needs a rebase. Could you do that?

Ack, will follow up with op. Thanks for taking action on this @justinchuby

**Comment by justinchuby** [comment]
Hi @xadupre could you maybe help with reviewing this or point us to someone who can? Thanks!

**Comment by NickGeneva** [comment]
Rebased! Let me know if theres issues. @justinchuby

**Comment by xadupre** [comment]
Is it possible to add a unit test to capture this change? It is possible to check the documentation reflects what the code says?

**Comment by NickGeneva** [comment]
@microsoft-github-policy-service agree [company="NVIDIA"]

**Comment by NickGeneva** [comment]
> Is it possible to add a unit test to capture this change? It is possible to check the documentation reflects what the code says?

Hi @xadupre 

Thank you for the comment, I've updated the documentation of this custom op with additional information. I have also updated the unit test to be one the appropriate even sized domain with a reference to the respective pytorch command the values were generated. 

Running `./onnxruntime_test_all` with a CUDA build passes on my system:
```
[----------] Global test environment tear-down
[==========] 4013 tests from 276 test suites ran. (220449 ms total)
[  PASSED  ] 4013 tests.
```

File: docs/ContribOperators.md [hunk 1]
```diff
@@ -2243,30 +2243,31 @@ This version of the operator has been available since version 1 of the 'com.micr
...
 
 This version of the operator has been available since version 1 of the 'com.microsoft' operator set.
+This function computes the inverse of the one-dimensional n-point RFFT computed in 'com.microsoft.rfft'.
 
 #### Attributes
 
 <dl>
 <dt><tt>normalized</tt> : int</dt>
-<dd></dd>
+<dd>must be 0, normalization currently not supported</dd>
```
Comment (justinchuby) [inline docs/ContribOperators.md:2252]: Is this op generated? Do we need an update to the op definition?

Comment (justinchuby) [reply docs/ContribOperators.md:2252]: https://github.com/microsoft/onnxruntime/blob/eeef15788804b0217d46ad6d1c2744585ad98d0c/onnxruntime/core/graph/contrib_ops/contrib_defs.cc#L1253-L1262 this seems to be the place

Comment (NickGeneva) [reply docs/ContribOperators.md]: Thanks for the catch @justinchuby  . I missed this. Updated the contrib_defs and reverted the markdown file.

The build still passes on my system.

**Comment by justinchuby** [comment]
Before the pipelines running just making sure you have run the file https://github.com/microsoft/onnxruntime/blob/main/tools/python/gen_contrib_doc.py and the generated result matches?

**Comment by NickGeneva** [comment]
Hi @justinchuby 

I just checked and seems this only generates some fields for the entire file (namely it does not generate any outputs for Inputs, outputs and general documentation for all operators). The fields that do get populated are fine.

I've installed ONNX from the CUDA pip wheel I built.

**Comment by justinchuby** [comment]
Looks like this https://github.com/microsoft/onnxruntime/blob/eeef15788804b0217d46ad6d1c2744585ad98d0c/tools/ci_build/build.py#L2184 is how it is being used in CI. Does it look like how you used it?

**Comment by NickGeneva** [comment]
Yep I run:
`python gen_contrib_doc.py  --output_path /workspace/gh-forks/onnx_temp/testdocs.md --domains com.microsoft` to produce the output file `testdocs.md`.

Maybe I need special flags set on my build?

Looking at the pipeline, the diffs seen look correct to me:
https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1078431&view=logs&jobId=7f366e99-16b2-52cc-e1ff-653af284e397&j=7f366e99-16b2-52cc-e1ff-653af284e397&t=5c9cf234-f957-5fdc-9c37-89899bf73c0c

**Comment by justinchuby** [comment]
I am not entirely sure. Worse case I would just apply the diff 😅

**Comment by NickGeneva** [comment]
Hi @justinchuby 

Seems two of the pipelines failed from the docs changing (expected and the diffs look like whats expected):
- [Windows GPU CI Pipeline](https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1078431&view=logs&j=7f366e99-16b2-52cc-e1ff-653af284e397&t=5c9cf234-f957-5fdc-9c37-89899bf73c0c)
- [Windows GPU CI Pipeline (kernelDocumentation build_x64_RelWithDebInfo) ](https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1078431&view=logs&j=7f366e99-16b2-52cc-e1ff-653af284e397&t=5c9cf234-f957-5fdc-9c37-89899bf73c0c)

I think the github runner that failed was due to some test just not running due to a test not getting built:

- [Windows_CI / Onnxruntime-TVM (pull_request)](https://github.com/microsoft/onnxruntime/actions/runs/5673130015/job/15379504437?pr=15662#step:8:18067)

Is there anything else that needs to be address?

**Comment by justinchuby** [comment]
Could you apply the diffs to the docs?

**Comment by NickGeneva** [comment]
@justinchuby 

Alright, just committed what should be the generated response. Hopefully did not miss any white space. Automatic doc generation is nice, but only when it works in a dev environment :( 

Hopefully the pipelines clear

**Comment by justinchuby** [comment]
Thanks! I agree this experience can be improved.

**Comment by NickGeneva** [comment]
Hi @justinchuby 

Seems some additional pipeline need to get ran? Getting this merged would be a great way to start the weekend if the stars align :)

**Comment by justinchuby** [comment]
CLA check is not passing. Not sure if its the bot...

**Comment by NickGeneva** [comment]
Sorry, whats the CLA? Is there something I need to do?

**Comment by NickGeneva** [comment]
@microsoft-github-policy-service agree company="NVIDIA"

**Comment by justinchuby** [comment]
> Sorry, whats the CLA? Is there something I need to do?
> 
> Oh do I need to get the original author of the PR to do the comment for license?

Yes. Or you may squash and rebase presuming you can sign all commits?

**Comment by Alexey-Kamenev** [comment]
@microsoft-github-policy-service agree company="NVIDIA"

**Comment by NickGeneva** [comment]
@justinchuby job done! (hopefully)

**Review by justinchuby — APPROVED** [review]

**Comment by justinchuby** [comment]
Have a nice weekend!

**Pull request merged**